### ISSUE TITLE
Error: Index was out of range.

### ISSUE THREAD

**Issue by goldbattle** [lay295/TwitchDownloader#224]
I also ran into this issue as in #221, but wanted to have an open issue.
Running 1.40.1
```
Unhandled exception. System.AggregateException: One or more errors occurred. (Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index'))
 ---> System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at System.Collections.Generic.List`1.get_Item(Int32 index)
   at TwitchDownloaderCore.ChatRenderer.<>c__DisplayClass6_1.<RenderVideoAsync>b__0()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location where exception was thrown ---
   at TwitchDownloaderCore.ChatRenderer.RenderVideoAsync(IProgress`1 progress, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at TwitchDownloaderCLI.Program.RenderChat(Options inputOptions)
   at TwitchDownloaderCLI.Program.Main(String[] args)
Aborted (core dumped)
```
With this VOD: https://www.twitch.tv/videos/1159913213

**Comment by lay295** [comment]
Yeah I messed up a fix I had for highlighted messages. I haven't had access to my computer for the past few days and will fix it when I'm back.

**Comment by goldbattle** [comment]
It looks like it fails on this line:
https://github.com/lay295/TwitchDownloader/blob/798f0dc97f90417b003189f611c71f8599958b5b/TwitchDownloaderCore/ChatRenderer.cs#L86
```
if (comment.message.user_notice_params.msg_id == "highlighted-message" && comment.message.fragments == null && comment.message.body != null)
{
   comment.message.fragments = new List<Fragment>();
   comment.message.fragments[0].text = comment.message.body;
}
```

Which means this might be related to #216
It also has two of these invalid highlight message which I believe is the issue:
```
{
   "_id": "b21b36d6-09e7-41c4-8ce1-08e2b48b8d6c",
   "created_at": "2021-09-26T19:11:50.143Z",
   "updated_at": "2021-09-26T19:11:50.143Z",
   "channel_id": "21841789",
   "content_type": "video",
   "content_id": "1159913213",
   "content_offset_seconds": 11553.443,
   "commenter":
   {
       "display_name": "l8o8gan",
       "_id": "252538386",
       "name": "l8o8gan",
       "type": "user",
       "bio": "Follow my stream I might be famous one day PogChamp ",
       "created_at": "2018-08-28T02:18:00.13544Z",
       "updated_at": "2021-09-26T19:11:08.53255Z",
       "logo": "https://static-cdn.jtvnw.net/jtv_user_pictures/cf0c0d26-19da-4770-ad4a-6f302b86a23f-profile_image-300x300.png"
   },
   "source": "chat",
   "state": "published",
   "message":
   {
       "body": "moonHype moonHype moonHype moonHype moonHype moonHype moonHype moonHype",
       "bits_spent": 0,
       "fragments": null,
       "is_action": false,
       "user_badges": null,
       "user_color": "#1E90FF",
       "user_notice_params":
       {
           "msg-id": "highlighted-message"
       },
       "emoticons":
       [
           {
               "_id": "81349",
               "begin": 1,
               "end": 8
           },
           {
               "_id": "81349",
               "begin": 10,
               "end": 17
           },
           {
               "_id": "81349",
               "begin": 19,
               "end": 26
           },
           {
               "_id": "81349",
               "begin": 28,
               "end": 35
           },
           {
               "_id": "81349",
               "begin": 37,
               "end": 44
           },
           {
               "_id": "81349",
               "begin": 46,
               "end": 53
           },
           {
               "_id": "81349",
               "begin": 55,
               "end": 62
           },
           {
               "_id": "81349",
               "begin": 64,
               "end": 71
           }
       ]
   },
   "more_replies": false
},
```
After I removed this comment, everything rendered fine.

**Comment by goldbattle** [comment]
I believe the if statement should be `comment.message.fragments != null`?

**Comment by lay295** [comment]
I just need to do something like this before setting the text property.
comment.message.fragments[0] = new Fragment();

I made the "fix" without actually testing if it worked because the fix seemed so straightforward. I'll fix it and #222 when I have my computer.
